src/frontend/running/farm.rs

```diff
@@ -274,15 +274,15 @@ impl FactoryComponent for FarmWidget {
                                 gtk::Image {
                                     set_icon_name: Some(icon_names::WARNING_OUTLINE),
                                     #[track = "self.changed_non_fatal_farming_error()"]
-                                    set_tooltip: {
+                                    set_tooltip: &{
                                         let last_error = self.non_fatal_farming_error
                                             .as_ref()
                                             .map(|error| error.to_string())
                                             .unwrap_or_default();
 
                                         T
                                             .running_farmer_farm_non_fatal_error_tooltip(last_error)
-                                            .as_str()
+                                            .as_str().to_string()
```

Comment:
This should also be possible to simplify the same way and even if not, please format this in a bit nicer way (it is not formatted automatically due to being inside of the macro).